sockeye/translate.py

```diff
@@ -78,27 +62,76 @@ def main():
                                                                                  args.models,
                                                                                  args.checkpoints,
                                                                                  args.softmax_temperature))
-        total_time = 0
-        i = 0
-        for i, line in enumerate(sys.stdin, 1):
-            trans_input = translator.make_input(i, line)
-            logger.debug(" IN: %s", trans_input)
-
-            tic = time.time()
-            trans_output = translator.translate(trans_input)
-            trans_wall_time = time.time() - tic
-            total_time += trans_wall_time
-
-            logger.debug("OUT: %s", trans_output)
-            logger.debug("OUT: time=%.2f", trans_wall_time)
-
-            output_handler.handle(trans_input, trans_output)
-
-        if i != 0:
-            logger.info("Processed %d lines. Total time: %.4f sec/sent: %.4f sent/sec: %.4f", i, total_time,
-                        total_time / i, i / total_time)
+        read_and_translate(translator, output_handler, args.source)
+
+
+def read_and_translate(translator: sockeye.inference.Translator, output_handler: sockeye.output_handler.OutputHandler,
+                       source: Optional[str] = None) -> None:
+    """
+
+    :param output_handler: Handler that will write output to a stream.
+    :param translator: Translator that will translate each line of input.
+    :param source: Path to file which will be translated line-by-line if included.
+    """
+
+    if source:
+        # Translate from a file.
+        i, total_time = translate_lines(output_handler, sockeye.data_io.smart_open(source), translator)
+    else:
+        # Translate from standard in, skip the first line of input.
+        i, total_time = translate_lines(output_handler, sys.stdin, translator, 1)
```

Comment:
ok, the enumerate(..., 1) actually means that we start counting at 1:
```
In [2]: list(enumerate([1,2]))
Out[2]: [(0, 1), (1, 2)]

In [3]: list(enumerate([1,2], 1))
Out[3]: [(1, 1), (2, 2)]```
I guess we don't need to do this differently for stdin vs from a file and just always start counting from 1.